build.gradle.kts

```diff
@@ -32,6 +32,7 @@ subprojects {
             kotlinOptions {
                 jvmTarget = "1.8"
                 allWarningsAsErrors = false
+                freeCompilerArgs = listOf("-Xjvm-default=all")
```

Comment:
per https://blog.jetbrains.com/kotlin/2020/07/kotlin-1-4-m3-generating-default-methods-in-interfaces/ I think this is the correct option

Reply:
This introduces a binary incompatibility. I think that might constitute a breaking change, but I am not too used to how java handles packages.

Reply:
fwiw tempest is [already using](https://github.com/squareup/cash-clientsync/pull/3244) default interface implementations similar to the one you added

Reply:
I confirmed that it fails to compile without this flag. I don't understand how this works with the other interfaces in the project, but if its required its required.

Reply:
I think the link above was a mis-paste?

I think the only way in which this introduces binary incompatibility is if we had existing usages of the deprecated `@JvmDefault` annotation with callers relying on the delegate. Since we don't have any usage of this annotation, I think this change is safe.

Reply:
Indeed, I was trying to link to [here](https://github.com/cashapp/tempest/blob/master/tempest2/src/main/kotlin/app/cash/tempest2/LogicalDb.kt#L105-L123).

I think you are right that this is safe. I wanted to avoid it if possible, since its just one more piece of awkwardness, but it does not appear to be avoidable here.